atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/SingleConnectionTransactionService.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2015 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.dbkvs.impl;
+
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.transaction.impl.TransactionConstants;
+import com.palantir.atlasdb.transaction.service.TransactionService;
+import com.palantir.common.base.Throwables;
+import com.palantir.exception.PalantirSqlException;
+import com.palantir.nexus.db.sql.BasicSQLUtils;
+import com.palantir.nexus.db.sql.PalantirSqlConnection;
+import com.palantir.sql.Connections;
+
+public final class SingleConnectionTransactionService implements TransactionService {
+    private final PalantirSqlConnection connection;
+
+    public SingleConnectionTransactionService(PalantirSqlConnection connection) {
+        this.connection = connection;
+    }
+
+    @Override
+    public Long get(long startTimestamp) {
+        throw new UnsupportedOperationException(
+                "Single connection transaction service for commit putUnlessExists only.");
+    }
+
+    @Override
+    public Map<Long, Long> get(Iterable<Long> startTimestamps) {
+        throw new UnsupportedOperationException(
+                "Single connection transaction service for commit putUnlessExists only.");
+    }
+
+    private static final String SQL_MET_INSERT_ONE_TRANSACTION =
+            "/* SQL_MET_INSERT_ONE ("+ TransactionConstants.TRANSACTION_TABLE.getQualifiedName() + ") */" +
+            " INSERT INTO pt_met_" + TransactionConstants.TRANSACTION_TABLE.getQualifiedName() +
+            " (row_name, col_name, ts, val) " +
+            " VALUES (?, ?, ?, ?) ";
+
+    @Override
+    public void putUnlessExists(long startTimestamp, long commitTimestamp) {
+        Cell cell = getTransactionCell(startTimestamp);
+        byte[] value = TransactionConstants.getValueForTimestamp(commitTimestamp);
+
+        try {
+            assert connection.getUnderlyingConnection().getAutoCommit() == false;
```

Comment:
nit: should probably extract this try/catch into a separate method to simplify and move assert to beginning of method as precondition.

```
        assert isAutoCommitDisabled() : "Auto commit was enabled when it should not have been.";

    private boolean isAutoCommitDisabled() {
        if (AssertUtils.isAssertEnabled()) {
            try {
                return connection.getUnderlyingConnection().getAutoCommit() == false;
            } catch (PalantirSqlException | SQLException e) {
                throw Throwables.throwUncheckedException(e);
            }
        }
        return true;
    }
```